diff --git a/Makefile.in b/Makefile.in
index 15739a5..27045d2 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1339,7 +1339,7 @@ convert.o: convert.c $(CONFIG_H) $(TREE_H) flags.h convert.h
 
 tree.o : tree.c $(CONFIG_H) $(TREE_H) flags.h function.h
 # CYGNUS LOCAL mentoropt/law
-shorten-life.o: shorten-life.c $(CONFIG_H) $(RTL_H) basic-block.h regs.h
+shorten-life.o: shorten-life.c $(CONFIG_H) $(RTL_H) basic-block.h regs.h insn-config.h
 # END CYGNUS LOCAL
 print-tree.o : print-tree.c $(CONFIG_H) $(TREE_H)
 stor-layout.o : stor-layout.c $(CONFIG_H) $(TREE_H) flags.h function.h
@@ -1385,7 +1385,7 @@ dbxout.o : dbxout.c $(CONFIG_H) $(TREE_H) $(RTL_H) flags.h regs.h \
    insn-config.h reload.h gstab.h xcoffout.h defaults.h output.h
 sdbout.o : sdbout.c $(CONFIG_H) $(TREE_H) $(RTL_H) flags.h except.h \
    function.h expr.h output.h hard-reg-set.h regs.h defaults.h real.h \
-   bytecode.h obstack.h xcoffout.h c-pragma.h
+   bytecode.h obstack.h xcoffout.h c-pragma.h insn-config.h
 dwarfout.o : dwarfout.c $(CONFIG_H) $(TREE_H) $(RTL_H) dwarf.h flags.h \
    insn-config.h reload.h output.h defaults.h
 dwarf2out.o : dwarf2out.c $(CONFIG_H) $(TREE_H) $(RTL_H) dwarf2.h flags.h \
