--- config/mips/mips.h	1995-06-15 19:32:41.000000000 +0000
+++ config/mips/mips-patched.h	2023-11-14 22:31:00.506591160 +0000
@@ -406,6 +406,9 @@
 			   | TARGET_ENDIAN_DEFAULT)}			\
 }

+/* Default little-endian */
+#define TARGET_ENDIAN_DEFAULT 0
+
 /* Default target_flags if no switches are specified  */

 #ifndef TARGET_DEFAULT
@@ -954,6 +957,22 @@
 #define ASM_OUTPUT_DESTRUCTOR(file, name)

 #endif /* 0 */
+
+/* A C statement to output something to the assembler file to switch to section
+   NAME for object DECL which is either a FUNCTION_DECL, a VAR_DECL or
+   NULL_TREE.  Some target formats do not support arbitrary sections.  Do not
+   define this macro in such cases.  */
+
+#define ASM_OUTPUT_SECTION_NAME(F, DECL, NAME) \
+do {								\
+  extern FILE *asm_out_text_file;				\
+  if ((DECL) && TREE_CODE (DECL) == FUNCTION_DECL)		\
+    fprintf (asm_out_text_file, "\t.section %s,\"ax\",@progbits\n", (NAME)); \
+  else if ((DECL) && TREE_READONLY (DECL))			\
+    fprintf (F, "\t.section %s,\"a\",@progbits\n", (NAME));	\
+  else								\
+    fprintf (F, "\t.section %s,\"aw\",@progbits\n", (NAME));	\
+} while (0)
 
 /* Target machine storage layout */

