--- obstack.h	1998-05-05 23:17:18.000000000 +0000
+++ obstack_patched.h	2023-08-14 20:03:25.162798764 +0000
@@ -417,7 +417,8 @@
 ({ struct obstack *__o = (OBSTACK);					\
    if (__o->next_free + sizeof (void *) > __o->chunk_limit)		\
      _obstack_newchunk (__o, sizeof (void *));				\
-   *((void **)__o->next_free)++ = ((void *)datum);			\
+   *((void **)__o->next_free) = ((void *)datum);			\
+   __o->next_free += sizeof (void *);	\
    (void) 0; })
 
 # define obstack_int_grow(OBSTACK,datum)				\
