diff --git a/obstack.h b/obstack.h
index dd8e861..e78fb0d 100644
--- a/obstack.h
+++ b/obstack.h
@@ -378,7 +378,8 @@ __extension__								\
    if (__o->next_free + sizeof (void *) > __o->chunk_limit)		\
      _obstack_newchunk (__o, sizeof (void *));				\
    if (!__o->alloc_failed)						\
-     *((void **)__o->next_free)++ = ((void *)datum);			\
+     *((void **)__o->next_free) = ((void *)datum);			\
+     __o->next_free += sizeof (void *);			\
    (void) 0; })
 
 #define obstack_int_grow(OBSTACK,datum)					\
@@ -387,7 +388,8 @@ __extension__								\
    if (__o->next_free + sizeof (int) > __o->chunk_limit)		\
      _obstack_newchunk (__o, sizeof (int));				\
    if (!__o->alloc_failed)						\
-     *((int *)__o->next_free)++ = ((int)datum);				\
+     *((int *)__o->next_free) = ((int)datum);				\
+     __o->next_free += sizeof (int);					\
    (void) 0; })
 
 #define obstack_ptr_grow_fast(h,aptr) (*((void **) (h)->next_free)++ = (void *)aptr)
