--- include/obstack.h	1998-09-05 12:25:19.000000000 +0000
+++ include/obstack_patched.h	2022-01-25 21:13:02.988292815 +0000
@@ -417,7 +417,8 @@
 ({ struct obstack *__o = (OBSTACK);					\
    if (__o->next_free + sizeof (void *) > __o->chunk_limit)		\
      _obstack_newchunk (__o, sizeof (void *));				\
-   *((void **)__o->next_free)++ = ((void *)datum);			\
+   *((void **)__o->next_free) = ((void *)datum);			\
+   __o->next_free += sizeof (void *);					\
    (void) 0; })

 # define obstack_int_grow(OBSTACK,datum)				\
